x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ldap/support/LdapMetadataResolver.java

```diff
@@ -54,12 +69,10 @@ public void resolve(
         TimeValue timeout,
         Logger logger,
         Collection<Attribute> attributes,
-        ActionListener<Map<String, Object>> listener
+        ActionListener<LdapMetadataResult> listener
     ) {
```

Comment:
I suggest we keep the behavior to not search the LDAP entry for attributes if both `user_full_name_attribute` and `user_email_attribute` are empty, like: `if (Strings.isEmpty(this.fullNameAttributeName) && Strings.isEmpty(this.emailAttributeName) && this.attributeNames.isEmpty()) { listener.onResponse(LdapMetadata.EMPTY); return; }`.

For the AD realm, in the default case, we're going to add an attribute search where there wasn't any before.
I think that's an OK decision weighting it against consistent experience with the LDAP realm, and the UX with profiles of LDAP/AD users. But still there must be a way to configure ES to avoid it completely.

Reply:
Ahh yes good idea. Added in 3aa03f50ab3